Adding troubleshoot entries

marctemp opened the pull request.

marctemp committed:
Adding troubleshoot entries

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+<br>
+
+What is the solution
++ Ensure the machine is connected to the Defra network (a VPN provides access to users off-site)
```
Technically it's not really the Defra network, it's the Azure vNET in the FFC subscription.  OpenVPN is also integrated with this vNET which is how we connect.

I'm not 100% sure whether the Defra corporate network, has access to Service Bus or not.  I've never tried.  Maybe I should...

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+
+What is the solution
++ Ensure the machine is connected to the Defra network (a VPN provides access to users off-site)
++ Check the environmental topic variables are correct for the service (these are usually stored locally within the user's profile file)
```
Slight nitpick but I'd may refer to them as environment variables for the service bus connection.  We don't know for definite how teams have set up their service bus environment variables (though most follow our approaches).  I'd also maybe mention `.env` files as a possibility as they're used to avoid collisions across environment variable names.

For example, say you've got a topic `TOPIC_ADDRESS` and you've got two subscribers where both repos refer to the subscription as `SUBSCRIPTION_ADDRESS`.  You couldn't add two different values in your profile file as one would just overwrite the other.  You'd need to add a `.env`  file to each repo so they can have their own values.

Hopefully the changes we've made to the SFI and payment services to use the `MESSAGE_SUFFIX` value will mean this one is less of a problem.

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
++ Ensure the machine is connected to the Defra network (a VPN provides access to users off-site)
++ Check the environmental topic variables are correct for the service (these are usually stored locally within the user's profile file)
++ (Stop)[https://docs.docker.com/engine/reference/commandline/stop/] and (prune all Docker containers, networks and volumes)[https://docs.docker.com/engine/reference/commandline/system_prune/]
++ Use this (testing framework)[https://github.com/johnwatson484/azure-service-bus-test-client] to check outbound requests independently of the application
```
() [] should be [] () to generate a link

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+
+What this looks like
++ In the application logs, `data: {"message": "Unable to create the amqp session due to operation timeout."}` or `ServiceBusError: Unable to create the amqp session due to operation timeout at ... code: 'ServiceTimeout'` or similar
+<br>
```
No need to add `<br>` if you've got either two spaces at the end of the previous line or a clear line break between lines

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+<br>
+
+What is the solution
++ Ensure you are looking at the correct Azure topic within the correct subscription
```
Nitpick: suggest using `-` rather than `+` just to be consistent with the other pages

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
++ Ensure you are looking at the correct Azure topic within the correct subscription
++ Check that the request does not (timeout)[#application-logs-message-or-service-timeout]
++ If the application is web-based, try again as webpack was possibly running
++ If the application is web-based, clear the web-browser's cache or use a (private/incongito window)[https://www.computerworld.com/article/3356840/how-to-go-incognito-in-chrome-firefox-safari-and-edge.html]
```
typo: incognito

johnwatson484 replied:
Link brackets also need reversed

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
++ Check (Azure platform's health)[https://status.azure.com/en-gb/status]
+<br>
+
+#### Messages in a topic's Service Bus Subscription(s) ae not processed
```
typo: `ae`

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+#### Messages in a topic's Service Bus Subscription(s) ae not processed
+
+What this looks like
++ On the Azure portal, the message count for the topic's Service Bus subcription(s) increase once a request has been sent
```
Typo: `subscription`

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+<br>
+
+What this means
++ The messages received from the Azure topic are pushed to the Service Bus scubscription(s) but not being processed
```
Typo: `subscription`

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+<br>
+
+What is the solution
++ Check a subcription is setup for the topic
```
Typo: `subscription`

johnwatson484 commented on guides/troubleshooting.md:
```diff
@@ -2,3 +2,54 @@
+What is the solution
++ Check a subcription is setup for the topic
++ Usually, another microservice is responsible for ingesting and processing these messages, ensure it is up
++ Check the environmental subscription variables are correct for the ingesting service (these are usually stored locally within the repository)
```
This equally applies to worker queues as well as the topic/subscription setup.  Maybe worth mentioning that although that will probably resonate with a reader anyway as it is reading this.

johnwatson484 commented:
Looks good, some useful tips in here.

Added a few suggestions, mostly formatting related.

If we covered all the Service Bus challenges we had, this page would probably go on forever.  But I think there is one more common one that jumps to mind that we should include.

by default Azure UI creates subscriptions that are auto deleted after 14 days of being idle.  Not great if you forget to uncheck a box and go on holiday.  What makes it even more frustrating is that client applications using service bus SDK don't fail fast, they just "pretend" they're connected without telling you the subscription no longer exists.  So maybe worth adding to the messages not received part?  Make sure the subscription is still there.

marctemp committed:
Added docker-compose v docker compose issue

marctemp committed:
Content updates

- Formatting and spelling mistakes fixed
- Details on vNET and .env files added

marctemp committed:
Added few suggestions
